### DISCUSSION TITLE
Assign pods to specific CPU slices

### DISCUSSION THREAD

**Discussion by HaoxinSEU** [k3s-io/k3s#11169] [Q&A] [answered]
Hi community,

I'm installing K3s on my machine. On my machine, I split my CPU into two slices `system.slice` and `user.slice` with different cores. In general `system.slice` is the OS, while `user.slice` is running applications.

I find out that K3s is running on the cores belonging to `system.slice`, and pods are also assigned to that slice. So I'm wondering how can I assign pods to `user.slice`?

I'm using Red Hat, and install K3s by running
```
curl -sfL https://get.k3s.io | sh -s - --write-kubeconfig-mode 644
```

Any hints would be much appreciated!

Best regards,
Haoxin

**Comment by brandond** [comment, marked as answer]
As per the systemd docs:
> By default, service and scope units are placed in system.slice, virtual machines and containers registered with systemd-machined are found in machine.slice, and user sessions handled by systemd-logind in user.slice.

Pods are not user sessions and don't belong in user.slice. If you put them there I suspect it might get confused?

**Reply by HaoxinSEU** [reply]
Thank you so much for your explanation, Brad!

So is there a way to not run pods on CPU cores in `system.slice`? For example, should I create a new slice for that?

Thanks again,
Haoxin

**Reply by brandond** [reply]
You might look at the `--cgroup-root` and `--cgroups-per-qos` kubelet args?

**Reply by HaoxinSEU** [reply]
Thanks Brad!

I will try this out :)
